### ISSUE TITLE
Things that would be nice to have with the functorch release

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#394]
The functorch release will be approximately 3/1. Below is a list of issues that would be good to resolve, ordered by priority.

## P0

Theme: important bugs and testing

- [x] https://github.com/pytorch/functorch/issues/381
- [x] https://github.com/pytorch/functorch/issues/393
- [x] https://github.com/pytorch/functorch/issues/382 
- [x] https://github.com/pytorch/pytorch/issues/70184
- [x] @zou3519 https://github.com/pytorch/functorch/issues/353 (warning: might be tricky) 
- [x] @samdow   loud error message for autograd.function and functorch in error case (https://github.com/pytorch/functorch/issues/207) 
- [x] @zou3519 Port NTK example to functorch
- [x] @zou3519 https://github.com/pytorch/functorch/issues/417
- [x] @samdow Make sure the examples still work 
- [x] @zou3519 https://github.com/pytorch/functorch/issues/360
- [ ] [Port https://github.com/srush/annotated-s4 to functorch](https://github.com/pytorch/functorch/issues/444)
- [x] [Evaluation on torchbenchmark models. E.g. (1) grad transform works on those, (2) compute per-sample-grads using vmap+grad, ...](https://github.com/pytorch/functorch/issues/445)
- [x] https://github.com/pytorch/functorch/issues/446
- [x] https://github.com/pytorch/functorch/issues/124
- [x] https://github.com/pytorch/functorch/issues/454

## P0.5
Theme: Other fixes and testing
- [ ] https://github.com/pytorch/functorch/issues/365
- [x] https://github.com/pytorch/functorch/issues/443
- [x] https://github.com/pytorch/functorch/issues/428
- [x] https://github.com/pytorch/functorch/issues/424
- [x] https://github.com/pytorch/functorch/issues/392
- [ ] https://github.com/pytorch/functorch/issues/342 (Just the performance testing parts of these, especially with jvp and jacfwd)
- [x] https://github.com/pytorch/functorch/issues/437
- [ ] Audit docs website
- [ ] @samdow Make batch norm error message more informative + write up how to fix models that use batch norm and run under vmap

## P1

It's OK if these don't make it into the release (we can do a follow-up later). But they are very nice to have.

- [x] @samdow Initial vmap randomness design 
- [x] https://github.com/pytorch/functorch/issues/384
- [ ] https://github.com/pytorch/functorch/issues/376 
- [ ] Switch make_functional over to nn.utils._stateless and/or clarify semantics (https://github.com/pytorch/functorch/issues/415)
- [x] https://github.com/pytorch/functorch/issues/451
